service/controller/resource/app/desired.go

```diff
@@ -145,17 +151,40 @@ func (r *Resource) newApp(cc controllercontext.Context, cr apiv1alpha2.Cluster,
 	}
 }
 
-func (r *Resource) newAppSpecs() []key.AppSpec {
-	switch r.provider {
-	case "aws":
-		return append(key.CommonAppSpecs(), key.AWSAppSpecs()...)
-	case "azure":
-		return append(key.CommonAppSpecs(), key.AzureAppSpecs()...)
-	case "kvm":
-		return append(key.CommonAppSpecs(), key.KVMAppSpecs()...)
-	default:
-		return key.CommonAppSpecs()
+func (r *Resource) newAppSpecs(ctx context.Context, cr apiv1alpha2.Cluster) ([]key.AppSpec, error) {
+	req := searcher.Request{
+		ReleaseVersion: key.ReleaseVersion(&cr),
+	}
+
+	res, err := r.clusterClient.Release.Searcher.Search(ctx, req)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	var specs []key.AppSpec
+	for _, app := range res.Apps {
+
+		spec := key.AppSpec{
+			App:             app.App,
+			Catalog:         appspec.Default.Catalog,
+			Chart:           fmt.Sprintf("%s-app", app.App),
+			Namespace:       appspec.Default.Namespace,
+			UseUpgradeForce: appspec.Default.UseUpgradeForce,
+			Version:         app.Version,
+		}
+		if val, ok := appspec.Exceptions[app.App]; ok {
+			if val.Chart != "" {
+				spec.Chart = val.Chart
+			}
+			if val.Namespace != "" {
+				spec.Namespace = val.Namespace
+			}
+			if !val.UseUpgradeForce {
+				spec.UseUpgradeForce = false
+			}
+		}
+		specs = append(specs, spec)
```

Comment:
A newline above would make this easier to read and I think matches our usual style.